tests/test_logic.py

```diff
@@ -89,9 +89,15 @@ def my_SOL(ij, sol):
     return (v if type(v) is bool else (True if v in sol else False) for v in ij)
 
 
+def _evaluate_eq(eq, sol):
+    if type(eq) is not dict:
+        eq = {c: v for v, c in eq if type(c) is not bool}
+    return sum(eq.get(s, 0) for s in sol if type(s) is not bool)
+
+
 def my_EVAL(eq, sol):
-    # evaluate_eq doesn't handle True/False entries
-    return evaluate_eq(eq, sol) + sum(c for c, a in eq if a is True)
+    # _evaluate_eq doesn't handle True/False entries
+    return _evaluate_eq(eq, sol) + sum(c for c, a in eq if a is True)
```

Comment:
Should this be an `is True` => `== TRUE`?

Reply:
Indeed! A latter commit took care of it ;).